### PR TITLE
fix(installer): authenticate release metadata lookups when a token is available

### LINKED ISSUE #6243: Installer canary failing
The public install paths (curl/PowerShell against install.opensre.com, GitHub release resolution, or Homebrew) failed a post-publish canary run.

Run: https://github.com/Tracer-Cloud/opensre/actions/runs/34795599846
Trigger: repository_dispatch
Pinned tag: v0.1.2026.9.14

Check the failed matrix leg(s) above and the uploaded `installer-canary-*` diagnostics artifact for the failing platform.

### PR DESCRIPTION
**Pull request by Davidson3556**
Fixes #6243

#### Describe the changes you have made in this PR -

The canary was failing on a release that had published correctly. Only the macOS leg failed, and only the two tests that touch `api.github.com`:

```
test_live_cdn_serves_install_sh          PASSED
test_live_install_sh_main_to_temp_dir    FAILED  curl: (56) ... error: 403
test_live_install_sh_release_channel     FAILED  curl: (56) ... error: 403
test_live_brew_tap_and_fetch             PASSED
test_live_brew_install_and_version       PASSED
```

`install.sh` resolves release metadata through the GitHub API, which allows **60 anonymous requests per hour per IP**. Runner egress is shared, so the macOS pool exhausts it and every metadata call returns 403 while the CDN and Homebrew paths, which do not use that API, keep passing. That is the whole failure: nothing was wrong with the build.

`install.sh` now sends a bearer token when `OPENSRE_GITHUB_TOKEN`, `GITHUB_TOKEN` or `GH_TOKEN` is set, and the canary job passes `secrets.GITHUB_TOKEN`. Anonymous installs behave exactly as before, which is the path real users take. Two deliberate constraints: the header is passed to curl through `--config` on stdin so the token cannot be read out of `ps`, and it is attached only to the metadata lookup in `download_text`, never to `download_to` where an asset download redirects off to `objects.githubusercontent.com`.

A 403 also surfaced as `Failed to query release metadata from GitHub.`, which gives a user behind a shared NAT nothing to act on. Anonymous callers now get the limit named along with the variable that lifts it; the hint is suppressed when a token is already set, so it cannot mislead.

### Demo/Screenshot for feature changes and bug fixes -

The mechanism, from this machine:

```
$ curl -sS -D- -o /dev/null https://api.github.com/rate_limit   # anonymous
x-ratelimit-limit: 60

anonymous:       limit=60   remaining=58
authenticated:   limit=5000 remaining=5000
```

The release itself was fine — `v0.1.2026.9.14` carries `opensre_0.1.2026.9.14_darwin-arm64.tar.gz` — and the exact failing command succeeds from an unshared IP on the same platform:

```
$ bash install.sh --release --install-dir <tmp> --version 0.1.2026.9.14
Checksum verification passed
OpenSRE v0.1.2026.9.14 installed successfully
```

Both paths still install after the change, run against a sandboxed `HOME`:

```
[anon] exit=0  installed=yes
[auth] exit=0  installed=yes
```

New error text, anonymous:

```
Error: Failed to query release metadata from GitHub. If the error above is HTTP 403,
this is usually GitHub's anonymous API limit of 60 requests per hour per IP address.
Set GITHUB_TOKEN to a token with public read access and run the installer again.
```

and with a token set, the hint is gone rather than trailing an empty string:

```
Error: Failed to query release metadata from GitHub.
```

Three regression tests in `tests/cli/test_install_sh_github_auth.py`. I checked they are not vacuous by reintroducing the insecure form (`-H "Authorization: ..."` on the command line) and confirming `test_token_reaches_curl_without_appearing_in_argv` fails, then restoring.

Checks: `make lint`, `make format-check`, `make typecheck`, and `uv run python -m pytest tests/cli/` (762 passed, 5 skipped).

### Left out, deliberately

`install.ps1` queries the same API anonymously and has the identical exposure. The Windows leg passed this time but can fail the same way. I did not change it because there is no `pwsh` on this machine, so I could not test it, and shipping an untested PowerShell change to the public installer is worse than leaving a known gap documented. Worth a follow-up.

The canary diagnostics artifact is also just a copy of the pytest log, so it adds nothing over the run output when triaging. Not touched here.

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

I started from the split in the results rather than the error text: everything touching `api.github.com` failed and everything else passed, at the same moment, on one platform only. That points at the API rather than the release, and the release assets confirmed it.

The alternative fix was retry with backoff. I rejected it: `curl --retry` does not retry 403 at all, so the existing `--retry 3` never applied here, and even a retry that did fire would be useless because the limit resets up to an hour later. Authentication is the only thing that changes the outcome.

The token is read from `OPENSRE_GITHUB_TOKEN` first to match the file's existing `OPENSRE_*` convention, then the two conventional names so CI and `gh`-configured shells work without extra setup.

Two things I was careful about. Putting the token in argv would expose it to any local `ps`, so it goes in through curl's config on stdin, and the test asserts that rather than trusting it. And attaching it to `download_to` would send credentials along an asset redirect to a different host, so only the metadata function carries it.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [ ] I have performed a self-review of my code
- [ ] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

**Comment by Davidson3556** [comment]
@codex review

File: tests/cli/test_install_sh_github_auth.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    bin_dir = _fake_curl(tmp_path)
+    script = textwrap.dedent(f"""\
+        __fn=$(awk '/^download_text\\(\\)/{{p=1}} p{{print}} p&&/^}}$/{{exit}}' {shlex.quote(str(INSTALL_SH))})
+        if [ -z "$__fn" ]; then
+            echo "download_text not found in install.sh" >&2
+            exit 1
+        fi
+        eval "$__fn"
+        CURL_FLAGS=(--fail --silent)
+        GITHUB_API_TOKEN={shlex.quote(token)}
```
[the remark this answers is not in the document]
Comment (Davidson3556) [reply tests/cli/test_install_sh_github_auth.py:53]: Right, and it was worse than a gap in coverage: I confirmed that misspelling GITHUB_TOKEN in the lookup, the one variable the canary passes, left all three tests green. Fixed in the latest push. The new tests eval the real assignment from install.sh with each public variable set and assert the header curl receives, plus precedence between them. Both were checked against a misspelt name and a reversed precedence before restoring.

**Comment by Davidson3556** [comment]
@greptile review

**Comment by muddlebee** [comment]
Closing this, as it doesn’t seem like a roadmap priority right now. We’re focusing on core stabilization first.
